hooks/unzipSound.js

```diff
@@ -17,48 +43,44 @@ module.exports = function(context) {
   } else {
     defer = context.requireCordovaModule("q").defer();
   }
-
   
   var platform = context.opts.plugin.platform;
   var platformConfig = utils.getPlatformConfigs(platform);
   if (!platformConfig) {
     utils.handleError("Invalid platform", defer);
   }
 
-  var sourcePath = platformConfig.getSoundSourceFolder()
-  var soundFolderPath = platformConfig.getSoundDestinationFolder()
+  var sourcePath = platformConfig.getSoundSourceFolder();
+  var soundFolderPath = platformConfig.getSoundDestinationFolder();
   
-  var soundZipFile = path.join(sourcePath, constants.soundZipFile)
-  if(fs.existsSync(soundZipFile)){
+  var soundZipFile = path.join(sourcePath, constants.soundZipFile);
+
+  if(utils.checkIfFileOrFolderExists(soundZipFile)){
     var zip = new AdmZip(soundZipFile);
     zip.extractAllTo(sourcePath, true);
     
-
-    var zipFolder = sourcePath + "/sounds"
-    if(!utils.checkIfFolderExists(zipFolder)){
-      utils.handleError("Something went wrong when trying to unzip sounds.zip", defer);
+    var entriesNr = zip.getEntries().length;
+    console.log("Number of entries in zip file: ", entriesNr);
+    
+    if(entriesNr == 0) {
+      utils.handleError("Sound zip file is empty, either delete it or add one or more files", defer);
       return
     }
-    var files = utils.getFilesFromPath(zipFolder);
 
-    if(!files){
-      utils.handleError("Something went wrong when trying to unzip sounds.zip", defer);
-      return
-    } 
-    
-    if(!utils.checkIfFolderExists(soundFolderPath)) {
-      utils.createOrCheckIfFolderExists(soundFolderPath)
-    }
+    var zipFolder = sourcePath + "/sounds"
 
-    for(var i = 0; i < files.length; i++) {
-      var filePath = path.join(zipFolder, files[i]);
-      var destFilePath = path.join(soundFolderPath, files[i]);
-      utils.copyFromSourceToDestPath(defer, filePath, destFilePath);
+    if(!utils.checkIfFileOrFolderExists(zipFolder)){
+      if(utils.isAndroid(platform))
+        copyWavFiles(platformConfig, sourcePath, soundFolderPath, defer)
+      utils.removeFile(soundZipFile);
```

Comment:
From my understanding this is used when the `utils.isAndroid(platform)` is true but also when it's false. Maybe we could reuse the same call?

Reply:
you're right! refactored it :D